src/Microsoft.Health.Fhir.SqlServer.UnitTests/Features/Search/Expressions/FlatteningRewriterTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Fhir.Core.Features.Search.Expressions;
+using Microsoft.Health.Fhir.SqlServer.Features.Search.Expressions.Visitors;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.SqlServer.UnitTests.Features.Search.Expressions
+{
+    public class FlatteningRewriterTests
+    {
+        [Fact]
+        public void GivenAMultiaryExpressionWithASingleElement_WhenFlattened_RemovesTheMultiary()
+        {
+            MultiaryExpression inputExpression = Expression.And(Expression.Equals(FieldName.Number, null, 1));
+            Expression visitedExpression = inputExpression.AcceptVisitor(FlatteningRewriter.Instance);
+            Assert.Equal("(FieldEqual Number 1)", visitedExpression.ToString());
```

Comment:
When we have time later, should we consider validating the actual output expressions instead of string since that removes the dependencies on the ToString implementation. (There are bunch of helpers in `SearchExpressionTestHelper` to help validating expression programmatically).

Reply:
We can discuss, in my opinion the test code is clearer when you compare strings rather than procedurally comparing object trees.